tests/posix/fs/src/test_fs_stat.c

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2023 Nikhef
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include <fcntl.h>
+#include <zephyr/posix/unistd.h>
+#include <zephyr/posix/dirent.h>
+#include "test_fs.h"
+
+#define FILL_SIZE 128
+
+#define TEST_FILE_SIZE     80
+#define TEST_DIR_FILE_SIZE 1000
+
+#define TEST_EMPTY_FILE FATFS_MNTP "/empty.dat"
+
+static void create_file(const char *filename, uint32_t size)
+{
+	int fh, res;
+
+	fh = open(filename, O_CREAT | O_WRONLY);
+	if (fh < 0) {
+		TC_ERROR("Failed opening file: %d (%s), errno=%d\n", fh, filename, errno);
+		__ASSERT_NO_MSG(fh >= 0);
+	}
+
+	uint8_t filling[FILL_SIZE];
+
+	while (size > FILL_SIZE) {
+		res = write(fh, filling, FILL_SIZE);
+		size -= FILL_SIZE;
+	}
+
+	res = write(fh, filling, size);
+	if (res < 0) {
+		TC_PRINT("Failed writing to file [%d]\n", res);
+		close(fh);
+		__ASSERT_NO_MSG(res == 0);
+	}
+
+	res = close(fh);
+	if (res < 0) {
+		TC_ERROR("Failed closing file: %d, errno=%d\n", res, errno);
+		__ASSERT_NO_MSG(res == 0);
+	}
+}
+
+static void *setup_fn(void)
+{
+	int res;
+
+	test_mount();
+
+	res = mkdir(TEST_DIR, 0070);
+	if (res < 0) {
+		TC_PRINT("Failed writing to creating dir [%d]\n", res);
+		__ASSERT_NO_MSG(res == 0);
+	}
+
+	create_file(TEST_FILE, TEST_FILE_SIZE);
+	create_file(TEST_DIR_FILE, TEST_DIR_FILE_SIZE);
+	create_file(TEST_EMPTY_FILE, 0);
+
+	return NULL;
+}
+
+static void teardown_fn(void *unused)
+{
+	ARG_UNUSED(unused);
+
+	int res;
+
+	res = unlink(TEST_FILE);
+	if (res < 0) {
+		TC_PRINT("Failed to unlink file " TEST_FILE " [%d]\n", res);
+		__ASSERT_NO_MSG(res == 0);
+	}
+	res = unlink(TEST_DIR_FILE);
+	if (res < 0) {
+		TC_PRINT("Failed to unlink file " TEST_DIR_FILE " [%d]\n", res);
+		__ASSERT_NO_MSG(res == 0);
+	}
+	res = unlink(TEST_DIR);
+	if (res < 0) {
+		TC_PRINT("Failed to unlink file " TEST_DIR " [%d]\n", res);
+		__ASSERT_NO_MSG(res == 0);
+	}
+
+	test_unmount(NULL);
+}
+
+ZTEST_SUITE(posix_fs_stat_test, NULL, setup_fn, NULL, NULL, teardown_fn);
```

Comment:
it might be better to convert these to before and after

Reply:
Afaik these functions which are called at the setup and teardown of the test-suite should be called accordingly. Though a quick search through the Zephyr code-base seems to indicate that nobody is really consistent with this. However, as you can see here: https://github.com/zephyrproject-rtos/zephyr/blob/68dc53b0779a37342f86a29362c8efdafdcea005/tests/bluetooth/audio/ascs/src/test_ase_state_transition_invalid.c#L67 they also use setup and teardown naming

Reply:
setup / teardown are only called once per suite and not once per test, so naming might be a bit confusing.

So the only reason I suggest before & after is so that there is no shared state between tests

Reply:
I can do it, but in principle (I know this is an assumption) there should only be read actions to the storage using the stat function. If the stat function does happen write to the storage it may also corrupt the underlying filesystem which is not reset between tests, as far as I can tell. I think we should also then clear the RAM before each mount operation, such that each tests indeed has a completely clean slate.

Reply:
Please see comments below.

Reply:
thanks!